### ISSUE TITLE
HDLC_Framer attempting to append to immutable Byte variable

### ISSUE THREAD

**Issue by Danieltajik** [daniestevez/gr-satellites#119]
Apologies for any naivety/ignorance on my part as this it the first issue I've ever filed on Git :).

When attempting to utilize the HDLC framer, the following error is thrown:

`buff.append(x)
AttributeError: 'bytes' object has no attribute 'append'
`

It seems that the buff variable is being set to the immutable bytes object. I believe this can be fixed by changing the following line in "hdlc_framer.py":

`        buff = hdlc.flag * self.preamble_bytes` 
to
`        buff = list(hdlc.flag * self.preamble_bytes)` 

As a sidebar, I would just like to say thank you for all your work Daniel, your repository and documentation has taught me immensely about the capabilities of gnuradio for use in satellite communications!

**Comment by daniestevez** [comment]
Thanks for the report!

It seems that I introduced this bug when porting the HDLC framer block from Python 2 to Python 3, and most likely nobody tested it since then. I have just fixed and also taken the opportunity to add a test for the HDLC framer and deframer blocks to try prevent this kind of things from happening (in general, gr-satellites lacks a lot of simple tests, but I'm planning to add some).
